src/main/java/io/jenkins/plugins/services/impl/HttpClientJiraIssues.java

```diff
@@ -0,0 +1,119 @@
+  }
+
+  private String getHttpContent(String url) {
+    url = this.configurationService.getJiraURL() + url;
+    logger.info("getHttpContent(" + url + ")");
+    final HttpGet get = new HttpGet(url);
+    try (final CloseableHttpClient httpClient = getHttpClient();
+         final CloseableHttpResponse response = httpClient.execute(get)) {
+      final HttpEntity entity = response.getEntity();
+      final String html = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+      EntityUtils.consume(entity);
+
+      if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+        return html;
+      } else {
+        final String msg = String.format("Unable to get content from %s - returned status code %d - %S", url, response.getStatusLine().getStatusCode(), html);
+        logger.warn(msg);
+        return null;
+      }
+    } catch (IOException e) {
+      final String msg = "Problem getting wiki content";
+      logger.error(msg, e);
+      return null;
+    }
+  }
+
+  public JiraIssues getIssues(String pluginName) throws IOException {
+    return getIssues(pluginName, 0);
+  }
+
+  public JiraIssues getIssues(String pluginName, int startAt) throws IOException {
+    int maxResults = 100;
+    JiraIssues jiraIssues = new JiraIssues();
+
+    String query = URLEncoder.encode("project=JENKINS AND status in (Open, \"In Progress\", Reopened) AND component=" + pluginName + "-plugin", "UTF-8");
+    String jsonInput = getHttpContent("/rest/api/2/search?startAt=" + startAt + "&maxResults=" + maxResults + "&jql=" + query);
+    if (Strings.isNullOrEmpty(jsonInput)) {
+      throw new IOException("Empty return value");
+    }
+
+    JSONObject obj = new JSONObject(jsonInput);
+
+    JSONArray jsonIssues = obj.getJSONArray("issues");
+    for (Object issue : jsonIssues) {
+      JSONObject jsonIssue = (JSONObject) issue;
+
+      JiraIssue jiraIssue = new JiraIssue();
+      jiraIssue.key = jsonIssue.getString("key");
+
+      JSONObject fields = jsonIssue.getJSONObject("fields");
+      jiraIssue.created = fields.optString("created");
+      jiraIssue.updated = fields.optString("updated");
+      jiraIssue.summary = fields.optString("summary");
+
+      JSONObject issuetype = fields.optJSONObject("issuetype");
+      JSONObject priority = fields.optJSONObject("priority");
+      JSONObject status = fields.optJSONObject("status");
+      JSONObject reporter = fields.optJSONObject("reporter");
+      JSONObject assignee = fields.optJSONObject("assignee");
+      JSONObject resolution = fields.optJSONObject("resolution");
+
+      jiraIssue.issueType = issuetype != null ? issuetype.optString("name") : null;
+      jiraIssue.priority =  priority != null ? priority.optString("name") : null;
+      jiraIssue.status =  status != null ? status.optString("status") : null;
+      jiraIssue.resolution = resolution != null ? reporter.optString("name") : null;
```

Comment:
is this the right field?

Reply:
Nope, thanks, that explains why i was having issue with it